test/shopify-cli/theme/dev_server/proxy_param_builder_test.rb

```diff
@@ -0,0 +1,121 @@
+  module Theme
+    module DevServer
+      class ProxyParamBuilderTest < Minitest::Test
+        def setup
+          super
+          @param_builder = ProxyParamBuilder.new
+        end
+
+        def test_empty_build
+          assert_equal({}, @param_builder.build)
+        end
+
+        def test_build_with_syncer_and_rack_env
+          @param_builder
+            .with_syncer(syncer)
+            .with_theme(theme)
+            .with_rack_env({ "HTTP_COOKIE" => http_cookie })
+
+          assert_equal({
+            "replace_templates[layout/theme.liquid]" => "<theme file content>",
+            "replace_templates[layout/password.liquid]" => "<password file content>",
+            "replace_templates[layout/config.json.json]" => "{ json file }",
+          }, @param_builder.build)
+        end
+
+        def test_build_with_syncer
+          @param_builder
+            .with_syncer(syncer)
+
+          assert_equal({
+            "replace_templates[layout/theme.liquid]" => "<theme file content>",
+            "replace_templates[layout/password.liquid]" => "<password file content>",
+            "replace_templates[layout/config.json.json]" => "{ json file }",
+          }, @param_builder.build)
+        end
+
+        def test_build_with_rack_env
+          @param_builder
+            .with_theme(theme)
+            .with_rack_env({ "HTTP_COOKIE" => http_cookie })
+
+          assert_equal({
+            "replace_templates[layout/theme.liquid]" => "<theme file content>",
+          }, @param_builder.build)
+        end
+
+        def test_build_with_rack_env_when_current_path_is_a_core_endpoint
+          @param_builder
+            .with_theme(theme)
+            .with_core_endpoints(["/core_end_point"])
+            .with_rack_env({
+              "HTTP_COOKIE" => http_cookie,
+              "PATH_INFO" => "/core_end_point",
+            })
+
+          assert_equal({}, @param_builder.build)
+        end
+
+        def test_build_with_rack_env_when_theme_is_nil
+          @param_builder
+            .with_rack_env({ "HTTP_COOKIE" => http_cookie })
+
+          assert_equal({}, @param_builder.build)
+        end
+
+        def test_build_when_rack_env_is_empty
+          @param_builder.with_rack_env({})
+
+          assert_equal({}, @param_builder.build)
+        end
+
+        private
+
+        def theme
+          stub("[]": liquid_file("theme"))
+        end
+
+        def liquid_file(name)
+          stub(
+            "read": "<#{name} file content>",
+            "relative_path": "layout/#{name}.liquid",
+            "liquid?": true,
+            "json?": false
+          )
+        end
+
+        def json_file(name)
+          stub(
+            "read": "{ json file }",
+            "relative_path": "layout/#{name}.json",
+            "liquid?": false,
+            "json?": true
+          )
+        end
+
+        def http_cookie
+          "cart_currency=EUR; storefront_digest=123; hot_reload_files=layout/theme.liquid"
```

Comment:
Not sure if this needs to change due to the comment above

Reply:
I've actually changed the `dev_server/hot-reload-theme.js` to use `hot_reload_files` everywhere :)